### PR TITLE
Feature/sup 11

### PR DESCRIPTION
**Pull request by FortuneAwaker**
Application vertical. Added needed entities, dtos and mappers. Changed methods activate and delete for base service: delete is left, activate and deactivate should be added to entity service interface if needed.

### CODE REVIEW

File: src/main/java/by/itech/lab/supplier/controller/ApplicationController.java [hunk 1]
```diff
@@ -1,13 +1,94 @@
...
+
+    @GetMapping(ApiConstants.URL_WAYBILL_ID_PARAMETER)
+    public Page<ApplicationDto> getAllByWaybill(@PathVariable Long waybillId,
+                                                       Pageable pageable) {
+        return applicationService.findAllByWayBill(pageable, waybillId);
+    }
+
+    @GetMapping(ApiConstants.URL_STATUS_PARAMETER)
+    public Page<ApplicationDto> getAllByWaybill(@PathVariable String status,
+                                                Pageable pageable) {
```
Comment (IrynaZazyba) [inline src/main/java/by/itech/lab/supplier/controller/ApplicationController.java]: maybe another name for this method?

File: src/main/java/by/itech/lab/supplier/controller/ApplicationController.java [hunk 2]
```diff
@@ -1,13 +1,94 @@
...
+    @PutMapping(ApiConstants.URL_ID_PARAMETER + ApiConstants.URL_STATUS_PARAMETER)
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void changeStatus(@PathVariable Long applicationId,
+                             @PathVariable String status) {
+        applicationService.changeStatus(applicationId, ApplicationStatus.valueOf(status));
+    }
+
+    @GetMapping(ApiConstants.URL_DELETED_PARAMETER)
+    public Page<ApplicationDto> getAllByDeleted(@PathVariable Boolean deleted,
+                                                Pageable pageable) {
```
Comment (IrynaZazyba) [inline src/main/java/by/itech/lab/supplier/controller/ApplicationController.java]: how we can we get deleted records

File: src/main/java/by/itech/lab/supplier/controller/CategoryController.java [hunk 3]
```diff
@@ -33,10 +32,10 @@ public CategoryDto save(@Valid @RequestBody CategoryDto categoryDto) {
         return categoryService.save(categoryDto);
     }
 
-    @GetMapping(ApiConstants.URL_ACTIVE_PARAMETER)
-    public Page<CategoryDto> getAllByActive(@PathVariable Boolean active,
-                                            Pageable pageable) {
-        return categoryService.findAllByActive(pageable, active);
+    @GetMapping(ApiConstants.URL_DELETED_PARAMETER)
+    public Page<CategoryDto> getAllByDeleted(@PathVariable Boolean deleted,
+                                             Pageable pageable) {
```
Comment (IrynaZazyba) [inline src/main/java/by/itech/lab/supplier/controller/CategoryController.java]: Why do we need deleted records?

File: src/main/java/by/itech/lab/supplier/domain/Address.java [hunk 4]
```diff
@@ -1,39 +1,40 @@
...
+import javax.persistence.Table;
 
-@Getter
-@Setter
+@Data
 @AllArgsConstructor
 @NoArgsConstructor
 @Entity
+@Builder
 @Table(name = "address")
```
Comment (IrynaZazyba) [inline src/main/java/by/itech/lab/supplier/domain/Address.java]: not necessary 'name'

File: src/main/java/by/itech/lab/supplier/domain/Application.java [hunk 5]
```diff
@@ -14,103 +23,32 @@
 @Builder
 @Entity
 @Table(name = "application")
```
Comment (IrynaZazyba) [inline src/main/java/by/itech/lab/supplier/domain/Application.java]: not necessary 'name'

File: src/main/java/by/itech/lab/supplier/domain/ApplicationStatus.java [hunk 6]
```diff
@@ -2,7 +2,7 @@
 
 public enum ApplicationStatus {
 
-    WAITING("Waiting"), IN_PROGRESS("In progress"), COMPLETED("Completed");
+    DEACTIVATED("Deactivated"), WAITING("Waiting"), IN_PROGRESS("In progress"), COMPLETED("Completed");
```
Comment (IrynaZazyba) [inline src/main/java/by/itech/lab/supplier/domain/ApplicationStatus.java]: are you sure that ApplicationStatus are like in TR?

File: src/main/java/by/itech/lab/supplier/domain/Car.java [hunk 7]
```diff
@@ -5,89 +5,37 @@
...
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
 
 @Data
 @AllArgsConstructor
 @NoArgsConstructor
 @Builder
 @Entity
 @Table(name = "car")
```
Comment (IrynaZazyba) [inline src/main/java/by/itech/lab/supplier/domain/Car.java]: not necessary 'name'

File: src/main/java/by/itech/lab/supplier/domain/WayBill.java [hunk 8]
```diff
@@ -6,136 +6,52 @@
...
 import java.sql.Date;
-import java.util.HashSet;
-import java.util.Set;
 
 @Data
 @AllArgsConstructor
 @NoArgsConstructor
 @Builder
 @Entity
 @Table(name = "waybill")
```
Comment (IrynaZazyba) [inline src/main/java/by/itech/lab/supplier/domain/WayBill.java:26]: not necessary 'name'

File: src/main/java/by/itech/lab/supplier/dto/mapper/ApplicationMapper.java [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+@Component
+@AllArgsConstructor
+public class ApplicationMapper implements BaseMapper<Application, ApplicationDto> {
+
+    private AddressMapper addressMapper;
+
+    private UserMapper userMapper;
+
+    private WayBillMapper wayBillMapper;
+
```
Comment (IrynaZazyba) [inline src/main/java/by/itech/lab/supplier/dto/mapper/ApplicationMapper.java:23]: final? remove extra lines

File: src/main/java/by/itech/lab/supplier/dto/mapper/CarMapper.java [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+
+
+@Component
+@AllArgsConstructor
+public class CarMapper implements BaseMapper<Car, CarDto> {
+
+    private CustomerMapper customerMapper;
+
+    private AddressMapper addressMapper;
+
```
Comment (IrynaZazyba) [inline src/main/java/by/itech/lab/supplier/dto/mapper/CarMapper.java:15]: final? remove extra lines

File: src/main/java/by/itech/lab/supplier/dto/mapper/WayBillMapper.java [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+@Component
+@AllArgsConstructor
+public class WayBillMapper implements BaseMapper<WayBill, WayBillDto> {
+
+    private AddressMapper addressMapper;
+
+    private CarMapper carMapper;
+
+    private UserMapper userMapper;
+
```
Comment (IrynaZazyba) [inline src/main/java/by/itech/lab/supplier/dto/mapper/WayBillMapper.java:16]: final? remove extra lines

File: src/main/java/by/itech/lab/supplier/repository/ApplicationRepository.java [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+
+import java.util.Optional;
+
+public interface ApplicationRepository extends JpaRepository<Application, Long> {
+
+    Optional<Application> findByNumber(final String number);
+
+    @Query("select app from Application app where :deleted is null or app.deleted=:deleted")
+    Page<Application> findAllByDeleted(Pageable pageable, @Param("deleted") Boolean deleted);
+
```
Comment (IrynaZazyba) [inline src/main/java/by/itech/lab/supplier/repository/ApplicationRepository.java]: see above

File: src/main/java/by/itech/lab/supplier/repository/ItemRepository.java [hunk 13]
```diff
@@ -15,17 +14,22 @@
...
     Page<Item> findAllByActive(Pageable pageable, @Param("active") Boolean active);
 
     @Modifying
     @Query("update Item set active = false where id = :id")
-    void delete(@Param("id") Long id);
+    void deactivate(@Param("id") Long id);
 
     @Modifying
     @Query("update Item set active = true where id = :id")
     void activate(@Param("id") Long id);
```
Comment (IrynaZazyba) [inline src/main/java/by/itech/lab/supplier/repository/ItemRepository.java]: 1 method can be used to activate and deactivate

File: src/main/java/by/itech/lab/supplier/repository/CategoryRepository.java [hunk 14]
```diff
@@ -13,14 +13,11 @@
 public interface CategoryRepository extends JpaRepository<Category, Long> {
     Optional<Category> findByCategory(final String categoryName);
 
-    @Query("select c from Category c where :active is null or c.active=:active")
-    Page<Category> findAllByActive(Pageable pageable, @Param("active") Boolean active);
+    @Query("select c from Category c where :deleted is null or c.deleted=:deleted")
+    Page<Category> findAllByDeleted(Pageable pageable, @Param("deleted") Boolean deleted);
```
Comment (IrynaZazyba) [inline src/main/java/by/itech/lab/supplier/repository/CategoryRepository.java]: see above

File: src/main/java/by/itech/lab/supplier/service/BaseService.java [hunk 15]
```diff
@@ -7,11 +7,10 @@
 public interface BaseService<Dto extends BaseDto> {
     Dto save(final Dto dto);
 
-    Page<Dto> findAllByActive(Pageable pageable, Boolean active);
+    Page<Dto> findAllByDeleted(Pageable pageable, Boolean deleted);
```
Comment (IrynaZazyba) [inline src/main/java/by/itech/lab/supplier/service/BaseService.java:8]: see above

File: src/main/java/by/itech/lab/supplier/service/ItemService.java [hunk 16]
```diff
@@ -11,4 +10,10 @@
 
     Page<ItemDto> findAllByCategory(String categoryName, Pageable pageable);
 
+    Page<ItemDto> findAllByActive(Pageable pageable, Boolean active);
+
+    void deactivate(final Long id);
+
+    void activate(final Long id);
+
```
Comment (IrynaZazyba) [inline src/main/java/by/itech/lab/supplier/service/ItemService.java]: 1 method can be used to activate and deactivate

File: src/main/java/by/itech/lab/supplier/service/impl/ApplicationServiceImpl.java [hunk 17]
```diff
@@ -0,0 +1,92 @@
...
+import java.util.Optional;
+
+@Service
+@AllArgsConstructor
+public class ApplicationServiceImpl implements ApplicationService {
+
+    private ApplicationRepository applicationRepository;
+
+    private ApplicationMapper applicationMapper;
+
```
Comment (IrynaZazyba) [inline src/main/java/by/itech/lab/supplier/service/impl/ApplicationServiceImpl.java:26]: final? remove extra lines

File: src/main/java/by/itech/lab/supplier/service/impl/ItemServiceImpl.java [hunk 18]
```diff
@@ -59,14 +59,24 @@ public ItemDto save(final ItemDto dto) {
         return itemMapper.map(saved);
     }
 
+    @Override
+    public Page<ItemDto> findAllByDeleted(Pageable pageable, Boolean deleted) {
+        return null;
+    }
+
```
Comment (IrynaZazyba) [inline src/main/java/by/itech/lab/supplier/service/impl/ItemServiceImpl.java:64]: null? see above

File: src/main/java/by/itech/lab/supplier/service/impl/ItemServiceImpl.java [hunk 19]
```diff
@@ -59,14 +59,24 @@ public ItemDto save(final ItemDto dto) {
...
     public ItemDto findById(final Long id) {
         return itemRepository.findById(id).map(itemMapper::map)
           .orElseThrow(NotFoundInDBException::new);
     }
 
+    @Override
+    public void delete(Long id) {
+
+    }
+
```
Comment (IrynaZazyba) [inline src/main/java/by/itech/lab/supplier/service/impl/ItemServiceImpl.java]: empty method

File: src/test/java/by/itech/lab/supplier/service/ApplicationServiceTest.java [hunk 20]
```diff
@@ -0,0 +1,192 @@
...
+        Mockito.when(applicationMapper.map(application)).thenReturn(applicationDto);
+
+        Assertions.assertEquals(applicationDto, applicationService.findByNumber("Test"));
+    }
+
+    @Test
+    void getApplicationByNumberTest_Negative() {
+        Mockito.when(applicationRepository.findByNumber("Test")).thenReturn(Optional.empty());
+
+
```
Comment (IrynaZazyba) [inline src/test/java/by/itech/lab/supplier/service/ApplicationServiceTest.java:116]: remove extra line

**Review by IrynaZazyba — COMMENTED** [review]
1. Need to bind items to the application
2.  According to TR: "A dispatcher can see both applications on item arrival to a warehouse and dispatching from a warehouse. A logistics specialist can see only applications for item dispatching from a warehouse."

File: src/main/java/by/itech/lab/supplier/dto/mapper/ApplicationMapper.java [hunk 21]
```diff
@@ -0,0 +1,65 @@
...
+          .sourceLocationAddressIdDto(addressMapper.map(application.getSourceLocationAddressId()))
+          .createdByUsersDto(userMapper.map(application.getCreatedByUsers()))
+          .lastUpdatedByUsersDto(userMapper.map(application.getLastUpdatedByUsers()))
+          .wayBillDto(wayBillMapper.map(application.getWayBill()))
+          .deleted(application.isDeleted())
+          .deletedAt(application.getDeletedAt())
+          .build();
+    }
+
+    public void update(final ApplicationDto from, final Application to) {
```
Comment (svareyko) [inline src/main/java/by/itech/lab/supplier/dto/mapper/ApplicationMapper.java]: please rename to map

File: src/main/java/by/itech/lab/supplier/dto/mapper/CarMapper.java [hunk 22]
```diff
@@ -0,0 +1,49 @@
...
+          .id(car.getId())
+          .totalCapacity(car.getTotalCapacity())
+          .currentCapacity(car.getCurrentCapacity())
+          .addressDto(addressMapper.map(car.getAddress()))
+          .number(car.getNumber())
+          .customerDto(customerMapper.map(car.getCustomer()))
+          .build();
+    }
+
+    public void update(final CarDto from, final Car to) {
```
Comment (svareyko) [inline src/main/java/by/itech/lab/supplier/dto/mapper/CarMapper.java]: please rename to map

File: src/main/java/by/itech/lab/supplier/dto/mapper/ZoneMapper.java [hunk 23]
```diff
@@ -0,0 +1,29 @@
...
+
+    @Override
+    public ZoneDto map(final Zone zone) {
+        return ZoneDto.builder()
+          .id(zone.getId())
+          .location(zone.getLocation())
+          .build();
+    }
+
+    public void update(final ZoneDto from, final Zone to) {
```
Comment (svareyko) [inline src/main/java/by/itech/lab/supplier/dto/mapper/ZoneMapper.java]: please rename to map

File: src/main/java/by/itech/lab/supplier/constant/ApiConstants.java [hunk 24]
```diff
@@ -2,6 +2,7 @@
 
 public interface ApiConstants {
```
Comment (eugenmolchanov) [inline src/main/java/by/itech/lab/supplier/constant/ApiConstants.java:4]: Use class as a container for constants, not an interface.

File: src/main/java/by/itech/lab/supplier/controller/ApplicationController.java [hunk 25]
```diff
@@ -1,13 +1,88 @@
...
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    //@Secured({"ROLE_DISPATCHER, ROLE_LOGISTICS_SPECIALIST"})
+    public void changeStatus(@PathVariable Long id,
+                             @PathVariable String status) {
+        applicationService.changeStatus(id, ApplicationStatus.valueOf(status));
+    }
+
+    @GetMapping
+    //@Secured({"ROLE_DISPATCHER, ROLE_LOGISTICS_SPECIALIST"})
+    public Page<ApplicationDto> getAll(final Pageable pageable) {
```
Comment (eugenmolchanov) [inline src/main/java/by/itech/lab/supplier/controller/ApplicationController.java:53]: usage of Pageable is mix of dao level with controller level

File: src/main/java/by/itech/lab/supplier/domain/Address.java [hunk 26]
```diff
@@ -2,40 +2,42 @@
...
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
 
+
+@Data
 @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
```
Comment (eugenmolchanov) [inline src/main/java/by/itech/lab/supplier/domain/Address.java:21]: It seems like this annotation should not be used here. It's dao level, not the controller.

File: src/main/java/by/itech/lab/supplier/domain/Application.java [hunk 27]
```diff
@@ -3,114 +3,64 @@
...
-    }
-
-    public void setDestinationLocationAddress(Address destinationLocationAddress) {
-        this.destinationLocationAddress = destinationLocationAddress;
-    }
-
+    private Address sourceLocationAddressId;
+    @ManyToOne
+    @JoinColumn(name = "destination_location_address_id")
+    private Address destinationLocationAddressId;
```
Comment (eugenmolchanov) [inline src/main/java/by/itech/lab/supplier/domain/Application.java]: Could be named as destinationLocationAddress

File: src/main/java/by/itech/lab/supplier/domain/Application.java [hunk 28]
```diff
@@ -3,114 +3,64 @@
...
     @JoinColumn(name = "source_location_address_id")
-    public Address getDestinationLocationAddress() {
-        return destinationLocationAddress;
-    }
-
-    public void setDestinationLocationAddress(Address destinationLocationAddress) {
-        this.destinationLocationAddress = destinationLocationAddress;
-    }
-
+    private Address sourceLocationAddressId;
```
Comment (eugenmolchanov) [inline src/main/java/by/itech/lab/supplier/domain/Application.java]: Could be named as sourceLocationAddress

File: src/main/java/by/itech/lab/supplier/domain/Address.java (hunk 26 above)
Comment (FortuneAwaker) [reply src/main/java/by/itech/lab/supplier/domain/Address.java:21]: +

File: src/main/java/by/itech/lab/supplier/dto/ApplicationDto.java [hunk 29]
```diff
@@ -0,0 +1,28 @@
...
+@Builder
+@Data
+public class ApplicationDto implements BaseDto {
+
+    Set<ItemsInApplicationDto> items = new HashSet<>();
+    private Long id;
+    private String number;
+    private LocalDate registrationDate;
+    private LocalDate lastUpdated;
+    private AddressDto sourceLocationAddressIdDto;
```
Comment (eugenmolchanov) [inline src/main/java/by/itech/lab/supplier/dto/ApplicationDto.java]: Could be named as sourceLocationAddressDto

File: src/main/java/by/itech/lab/supplier/dto/ApplicationDto.java [hunk 30]
```diff
@@ -0,0 +1,28 @@
...
+    private AddressDto destinationLocationAddressIdDto;
```
Comment (eugenmolchanov) [inline src/main/java/by/itech/lab/supplier/dto/ApplicationDto.java]: Could be named as destinationLocationAddressDto

File: src/main/java/by/itech/lab/supplier/dto/mapper/ApplicationMapper.java [hunk 31]
```diff
@@ -0,0 +1,116 @@
...
+        to.setCreatedByUsers(userMapper.map(from.getCreatedByUsersDto()));
+        to.setLastUpdatedByUsers(userMapper.map(from.getLastUpdatedByUsersDto()));
+        to.setWayBill(wayBillMapper.map(from.getWayBillDto()));
+        to.setDeletedAt(from.getDeletedAt());
+        updateItems(to.getItems(), from.getItems());
+    }
+
+    public Application mapItems(final Application application) {
+        for (ItemsInApplication item : application.getItems()
+        ) {
```
Comment (eugenmolchanov) [inline src/main/java/by/itech/lab/supplier/dto/mapper/ApplicationMapper.java]: Please, format this method body

File: src/main/java/by/itech/lab/supplier/dto/mapper/ApplicationMapper.java [hunk 32]
```diff
@@ -0,0 +1,116 @@
...
+            item.setApplication(application);
+        }
+        return application;
+    }
+
+    private void updateItems(Set<ItemsInApplication> forUpdate, Set<ItemsInApplicationDto> update) {
+        for (ItemsInApplicationDto item : update
```
Comment (eugenmolchanov) [inline src/main/java/by/itech/lab/supplier/dto/mapper/ApplicationMapper.java]: Please, format this method body

File: src/main/java/by/itech/lab/supplier/domain/Application.java (hunk 27 above)
Comment (FortuneAwaker) [reply src/main/java/by/itech/lab/supplier/domain/Application.java]: +

File: src/main/java/by/itech/lab/supplier/domain/Application.java (hunk 28 above)
Comment (FortuneAwaker) [reply src/main/java/by/itech/lab/supplier/domain/Application.java]: +++

File: src/main/java/by/itech/lab/supplier/dto/ApplicationDto.java (hunk 29 above)
Comment (FortuneAwaker) [reply src/main/java/by/itech/lab/supplier/dto/ApplicationDto.java]: +++

File: src/main/java/by/itech/lab/supplier/dto/ApplicationDto.java (hunk 30 above)
Comment (FortuneAwaker) [reply src/main/java/by/itech/lab/supplier/dto/ApplicationDto.java]: +++

File: src/main/java/by/itech/lab/supplier/service/impl/ApplicationServiceImpl.java [hunk 33]
```diff
@@ -0,0 +1,84 @@
...
+@Service
+@AllArgsConstructor
+public class ApplicationServiceImpl implements ApplicationService {
+
+    private final ApplicationRepository applicationRepository;
+
+    private final ApplicationMapper applicationMapper;
+
+    @Override
+    public ApplicationDto save(final ApplicationDto dto) {
```
Comment (eugenmolchanov) [inline src/main/java/by/itech/lab/supplier/service/impl/ApplicationServiceImpl.java:29]: Must be Transactional

File: src/main/java/by/itech/lab/supplier/service/impl/CategoryServiceImpl.java [hunk 34]
```diff
@@ -24,22 +25,24 @@
...
     @Override
-    public Page<CategoryDto> findAllByActive(final Pageable pageable, final Boolean active) {
-        return categoryRepository.findAllByActive(pageable, active)
+    public Page<CategoryDto> findAll(final Pageable pageable) {
+        return categoryRepository.findAll(pageable)
           .map(categoryMapper::map);
     }
 
+    @Override
     public CategoryDto save(final CategoryDto dto) {
```
Comment (eugenmolchanov) [inline src/main/java/by/itech/lab/supplier/service/impl/CategoryServiceImpl.java:40]: Must be Transactional

File: src/main/java/by/itech/lab/supplier/service/impl/ItemServiceImpl.java [hunk 35]
```diff
@@ -27,30 +28,25 @@
...
           .map(itemMapper::map);
     }
 
     @Override
-    public Page<ItemDto> findAllByActive(final Pageable pageable, final Boolean active) {
-        return itemRepository.findAllByActive(pageable, active)
-          .map(itemMapper::map);
-    }
-
     public ItemDto save(final ItemDto dto) {
```
Comment (eugenmolchanov) [inline src/main/java/by/itech/lab/supplier/service/impl/ItemServiceImpl.java:44]: Must be Transactional

**Review by eugenmolchanov — CHANGES_REQUESTED** [review]
Please, see the comments

File: src/main/java/by/itech/lab/supplier/constant/ApiConstants.java (hunk 24 above)
Comment (FortuneAwaker) [reply src/main/java/by/itech/lab/supplier/constant/ApiConstants.java:4]: +++

File: src/main/java/by/itech/lab/supplier/controller/ApplicationController.java (hunk 25 above)
Comment (FortuneAwaker) [reply src/main/java/by/itech/lab/supplier/controller/ApplicationController.java:53]: +++

File: src/main/java/by/itech/lab/supplier/dto/mapper/ApplicationMapper.java (hunk 31 above)
Comment (FortuneAwaker) [reply src/main/java/by/itech/lab/supplier/dto/mapper/ApplicationMapper.java]: +++

File: src/main/java/by/itech/lab/supplier/dto/mapper/ApplicationMapper.java (hunk 32 above)
Comment (FortuneAwaker) [reply src/main/java/by/itech/lab/supplier/dto/mapper/ApplicationMapper.java]: +++

File: src/main/java/by/itech/lab/supplier/service/impl/ApplicationServiceImpl.java (hunk 33 above)
Comment (FortuneAwaker) [reply src/main/java/by/itech/lab/supplier/service/impl/ApplicationServiceImpl.java:29]: +++

File: src/main/java/by/itech/lab/supplier/service/impl/CategoryServiceImpl.java (hunk 34 above)
Comment (FortuneAwaker) [reply src/main/java/by/itech/lab/supplier/service/impl/CategoryServiceImpl.java:40]: +++

File: src/main/java/by/itech/lab/supplier/service/impl/ItemServiceImpl.java (hunk 35 above)
Comment (FortuneAwaker) [reply src/main/java/by/itech/lab/supplier/service/impl/ItemServiceImpl.java:44]: +++

File: src/main/java/by/itech/lab/supplier/domain/Address.java (hunk 26 above)
Comment (FortuneAwaker) [reply src/main/java/by/itech/lab/supplier/domain/Address.java:21]: +++

File: src/main/java/by/itech/lab/supplier/domain/Application.java (hunk 27 above)
Comment (FortuneAwaker) [reply src/main/java/by/itech/lab/supplier/domain/Application.java]: +++

**Review by eugenmolchanov — APPROVED** [review]

File: src/main/java/by/itech/lab/supplier/dto/mapper/CarMapper.java [hunk 36]
```diff
@@ -0,0 +1,53 @@
...
+
+
+@Component
+public class CarMapper implements BaseMapper<Car, CarDto> {
+
+    private final CustomerMapper customerMapper;
+    private final AddressMapper addressMapper;
+
+    @Autowired
+    public CarMapper(final CustomerMapper customerMapper, final AddressMapper addressMapper) {
```
Comment (svareyko) [inline src/main/java/by/itech/lab/supplier/dto/mapper/CarMapper.java]: why you dont use lombok?

File: src/main/java/by/itech/lab/supplier/dto/mapper/ItemMapper.java [hunk 37]
```diff
@@ -2,41 +2,49 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
 @Component
 public class ItemMapper implements BaseMapper<Item, ItemDto> {
 
+    private final CategoryMapper categoryMapper;
+
+    @Autowired
+    public ItemMapper(final CategoryMapper categoryMapper) {
```
Comment (svareyko) [inline src/main/java/by/itech/lab/supplier/dto/mapper/ItemMapper.java]: lombok?

File: src/main/java/by/itech/lab/supplier/dto/mapper/ItemsInApplicationMapper.java [hunk 38]
```diff
@@ -0,0 +1,47 @@
...
+import org.springframework.stereotype.Component;
+
+
+@Component
+public class ItemsInApplicationMapper implements BaseMapper<ItemsInApplication, ItemsInApplicationDto> {
+
+    private final ItemMapper itemMapper;
+
+    @Autowired
+    public ItemsInApplicationMapper(final ItemMapper itemMapper) {
```
Comment (svareyko) [inline src/main/java/by/itech/lab/supplier/dto/mapper/ItemsInApplicationMapper.java]: lombok?

File: src/main/java/by/itech/lab/supplier/dto/mapper/WayBillMapper.java [hunk 39]
```diff
@@ -0,0 +1,54 @@
...
+@Component
+public class WayBillMapper implements BaseMapper<WayBill, WayBillDto> {
+
+    private final AddressMapper addressMapper;
+    private final CarMapper carMapper;
+    private final UserMapper userMapper;
+
+    @Autowired
+    public WayBillMapper(final AddressMapper addressMapper, final CarMapper carMapper, final UserMapper userMapper) {
```
Comment (svareyko) [inline src/main/java/by/itech/lab/supplier/dto/mapper/WayBillMapper.java]: lombok?

File: src/main/java/by/itech/lab/supplier/dto/mapper/CarMapper.java (hunk 36 above)
Comment (FortuneAwaker) [reply src/main/java/by/itech/lab/supplier/dto/mapper/CarMapper.java]: +++

File: src/main/java/by/itech/lab/supplier/dto/mapper/ItemMapper.java (hunk 37 above)
Comment (FortuneAwaker) [reply src/main/java/by/itech/lab/supplier/dto/mapper/ItemMapper.java]: +++

File: src/main/java/by/itech/lab/supplier/dto/mapper/ItemsInApplicationMapper.java (hunk 38 above)
Comment (FortuneAwaker) [reply src/main/java/by/itech/lab/supplier/dto/mapper/ItemsInApplicationMapper.java]: +++

File: src/main/java/by/itech/lab/supplier/dto/mapper/WayBillMapper.java (hunk 39 above)
Comment (FortuneAwaker) [reply src/main/java/by/itech/lab/supplier/dto/mapper/WayBillMapper.java]: +++

**Review by svareyko — APPROVED** [review]

**Pull request merged**